@AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ihadss/fn_ihadssDraw.sqf

```diff
@@ -728,33 +728,29 @@ if (_heli getVariable "fza_ah64_hmdfsmode" != "cruise") then {
 _bobcoords = [-100, -100];
 if (_heli getVariable "fza_ah64_hmdfsmode" == "bobup") then {
 
-    _thetabob = (360 + ((_heli getVariable "fza_ah64_bobhdg") - direction _heli)) Mod 360;
-
-    if (_thetabob >= 180) then {
-        _thetabob = _thetabob - 360;
-    } else {
-        _thetabob = _thetabob;
-    };
-
-    _curwpdir = _thetabob;
-
-    _bobcoordsx = (((_heli getVariable "fza_ah64_bobpos" select 0) - (getposasl _heli select 0)) * 0.017) + 0.480775;
-    if (_bobcoordsx > 0.7) then {
-        _bobcoordsx = 0.7;
-    };
-    if (_bobcoordsx < 0.3) then {
-        _bobcoordsx = 0.3;
-    };
-    _bobcoordsy = (((getposasl _heli select 1) - (_heli getVariable "fza_ah64_bobpos" select 1)) * 0.017) + 0.475;
-    if (_bobcoordsy > 0.7) then {
-        _bobcoordsy = 0.7;
-    };
-    if (_bobcoordsy < 0.3) then {
-        _bobcoordsy = 0.3;
-    };
+    _BobMultiGraph =  [[0,       0],
+                       [6.096,  0.17]];
+
+    _thetabob = (360 + ((_heli getdir (_heli getVariable "fza_ah64_bobpos")) - direction _heli)) Mod 360;
+    _HeliBobDist = (_heli distance2d (_heli getVariable "fza_ah64_bobpos"));
+    _distance = [_BobMultiGraph, _HeliBobDist] call fza_fnc_linearInterp;
+    _CoordX = (sin _thetabob);
+    _CoordY = (cos _thetabob);
```

Comment:
A large number of these variables have not been declared properly. If you are declaring a variable for use for the first time, you must use `private _varName =`